### DISCUSSION TITLE
"Justify" last row of bars

### DISCUSSION THREAD

**Discussion by BenTalagan** [CoderLine/alphaTab#1230] [Ideas]
Hello and thank you for the hard work done on AlphaTab. This is truly impressive !

One option I am missing and could not find (the website is down these last days and it's harder to dig for info) is the ability to justify the last row of bars on a score. As far as I understand, the upcoming version 1.3.0 is now able to parse the page layout given in the Guitar Pro file which is completely amazing ; however, independently, it'd be great to be able to align the last bar of the last row of bars with the right side of the score for aesthetics reasons and profit of the remaining white space at the end. This is especially true for small snippets of one or two rows to which you would like to give some air by using the full width of the container.

I hope this description is clear and relevant and thanks again :)

**Comment by Danielku15** [comment]
The website is online again. Not sure why it went down, some previous deploy of an update failed unexpectedly. 

It is true that for the last row there is no justification done. The related code is [here](https://github.com/CoderLine/alphaTab/blob/b46162aab3f7aed96f2989ff6e0e145941ae61a6/src/rendering/layout/PageViewLayout.ts#L348-L356). Adding a new setting should be very straight forward. 

I added https://github.com/CoderLine/alphaTab/issues/1233 as backlog item.

**Reply by BenTalagan** [reply]
Thank you Daniel for your fast answer. Can't wait to see the next improvements !

**Comment by Danielku15** [comment]
@BenTalagan  There it is: https://github.com/CoderLine/alphaTab/pull/1240 Will become available with the build this night (1.3.0-alpha.720)

https://next.alphatab.net/docs/reference/settings/display/justifylastsystem/

**Reply by BenTalagan** [reply]
Awesome ! Many thanks for the hard work.
